lib/to_do/handlers/to_do.ex

```diff
@@ -17,4 +17,10 @@ defmodule ToDo.Handlers.ToDo do
       _ -> value
     end
   end
+
+  def handle(%Request{method: "DELETE", full_path: [_path, id | _]} = _req) do
+    [key | _] = id |> String.split("/")
```

Comment:
This looks like it has the potential for reuse, and the implementation may even change before this project is complete.  Should this move to its own function somewhere? or should the id be accesable on the Request? (ex: `def handle(%Request{method: "DELETE", id: id}) do` ...)

Reply:
That's a good point! I'm willing to move this parsing into the API class, and instead pass the entire id path list `["/todo", "/:id" ]` down to `API.delete/1`. I didn't want to be too specific though, because I was worried about violating the open-closed principle if I decided to keep building out. i.e., what if I ever want to mount another path on /todo? In the specific instance of`/todo/delete/1` I now have to rewrite Request because of how it chooses to store the id value.

Reply:
read up on HTTP path requirements form the spec, to understand what is expected and supported.  I do not think you example (`/todo/delete/1`) actually is, and for this project we are working to spec.